raft-tech/TANF-app
# 2408.1 - lift and shift search_indexes.parsers to parsers

## Summary of Changes
As part of #2408, move the `tdpservice.search_indexes.parsers` module into its own django app `tdpservice.parsers`. This required only a few changes
- Create a new django app with an `apps.py` that didn't exist before.
- Move all the models, views, parsers, and utils to the new app, creating a couple migrations to handle the `ParserError` model moving locations.
- Update `settings.common` to utilize the new parsers app.
- Update some imports.

The rest of 2408 will be completed in additional PRs.

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && npm run start
cd tdrs-backend && docker-compose up
```

1. Open http://localhost:3000/ and sign in.
1. Upload a DataFile and observe that the relevant information is parsed.
1. Run automated tests.

## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] Move parsing to its own app and reorganize folder structure (tdpservice/search_indexes/parsers --> tdpservice/parsers)

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
